### PR TITLE
feat: Add pages - add cluster, check local requirements, install local cluster, add remote cluster and connecto to remote cluster

### PR DESCRIPTION
**Pull request by mgs95**

### CODE REVIEW

File: kdl-client/electron/check_requirements.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const electron = require('electron');
+const { exec } = require('child_process');
+
+const commands = {
+  k8s: 'kubectl get nodes',
```
Comment (Sergiodfdez) [inline kdl-client/electron/check_requirements.js:5]: In this case you must use this command to get at least one ready node, the command is incomplete.
```bash
kubectl get nodes -o jsonpath='{.items[*].status.conditions[*].type}'
```

File: kdl-client/src/Constants/routes.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const ROUTE = {
+  HOME: '/',
+  NEW_CLUSTER: '/',
```
Comment (Sergiodfdez) [inline kdl-client/src/Constants/routes.ts:3]: Is the same route HOME that NEW_CLUSTER ?

File: kdl-client/src/Pages/InstallLocalCluster/InstallLocalCluster.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    // We want to execute this on on component mount/unmount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let buttons;
+  switch (installationState) {
+    case InstallationState.OK:
+      buttons = (
+        <>
+          <Button label="CONNECT NOW" to={'/'} primary />
```
Comment (Sergiodfdez) [inline kdl-client/src/Pages/InstallLocalCluster/InstallLocalCluster.tsx:84]:
```suggestion
          <Button label="CONNECT NOW" to={ROUTE.HOME} primary />
```

**Review by Sergiodfdez — CHANGES_REQUESTED** [review]

File: kdl-client/electron/install_local_cluster.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  event.sender.send('installLocalClusterReply', {
+    finished: true,
+    isError: !success,
+    text: `child process exited with code ${code}`,
+  });
+
+  return success;
+}
+
+function createNameSpace(event) {
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/electron/install_local_cluster.js]: Changed

File: kdl-client/electron/install_local_cluster.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    cmd.stdout.on('data', data => onData(event, data));
+    cmd.stderr.on('data', data => onError(event, data));
+    cmd.on('close', (code) => resolve(onClose(code, event)));
+  });
+}
+
+console.log('Add installLocalCluster listener...');
+electron.ipcMain.on('installLocalCluster', event => {
+  createNameSpace(event).then(sucess => {
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/electron/install_local_cluster.js]: Changed

File: kdl-client/electron/install_local_cluster.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    cmd.on('close', code => resolve(onClose(code, event)));
+  });
+}
+
+function installComponents(event) {
+  return new Promise(resolve => {
+    const cmd = spawn('helm', [
+      'upgrade',
+      '--install',
+      'konstellation-local',
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/electron/install_local_cluster.js]: Changed to kdl

File: kdl-client/electron/install_local_cluster.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      '--namespace',
+      'kdl-local',
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/electron/install_local_cluster.js]: Changed to kdl

File: kdl-client/package.json [hunk 8]
```diff
@@ -23,9 +23,11 @@
     "electron-is-dev": "^1.2.0",
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/package.json]: Removed

File: kdl-client/src/Pages/NewCluster/NewCluster.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+          <ClusterOption
+            title="Install a Local Cluster"
+            subtitle="Fusce vehicula dolor arcu, sit amet blandit dolor mollis nec. Donec viverra eleifend lacus, vitae ullamcorper metus. Sed sollicitudin ipsum quis nunc sollicitudin ultrices."
+            actionLabel="INSTALL"
+            to={ROUTE.CHECK_LOCAL_CLUSTER_REQUIREMENTS}
+            Cluster={<Cluster direction={DIRECTION.DOWN} />}
+          />
+        </div>
+        <div className={styles.cancel}>
+          <Button label="CANCEL" to="/" />
```
[the remark this answers is not in the document]
Comment (mgs95) [reply kdl-client/src/Pages/NewCluster/NewCluster.tsx]: Changed

File: kdl-client/electron/check_requirements.js (hunk 1 above)
Comment (mgs95) [reply kdl-client/electron/check_requirements.js]: Updated

File: kdl-client/src/Constants/routes.ts (hunk 2 above)
Comment (mgs95) [reply kdl-client/src/Constants/routes.ts]: Changed

File: kdl-client/src/Pages/InstallLocalCluster/InstallLocalCluster.tsx (hunk 3 above)
Comment (mgs95) [reply kdl-client/src/Pages/InstallLocalCluster/InstallLocalCluster.tsx]: Changed

**Review by Sergiodfdez — APPROVED** [review]

**Pull request merged**